src/status/wallet/collectibles.nim

```diff
@@ -0,0 +1,116 @@
+import strformat, httpclient, json, chronicles, sequtils, strutils, tables
+import ../libstatus/core as status
+import ../libstatus/contracts as contracts
+import eth/common/eth_types
+
+type Collectible* = ref object
+    name*, image*: string
+
+proc getTokenUri(contract: Contract, tokenId: int): string =
+  try:
+    let payload = %* [{
+      "to": $contract.address,
+      "data": contract.methods["tokenURI"].encodeAbi(tokenId)
+    }, "latest"]
+    let response = status.callPrivateRPC("eth_call", payload)
+    var postfixedResult: string = parseJson($response)["result"].str
+    postfixedResult.removeSuffix('0')
+    postfixedResult.removePrefix("0x")
+    postfixedResult = parseHexStr(postfixedResult)
+    let index = postfixedResult.find("http")
+    if (index < -1):
+      return ""
+    result = postfixedResult[index ..  postfixedResult.high]
+  except Exception as e:
+    error "Error getting the token URI", mes = e.msg
+    result = ""
+
+proc tokenOfOwnerByIndex(contract: Contract, address: EthAddress, index: int): int =
+  let payload = %* [{
+    "to": $contract.address,
+    "data": contract.methods["tokenOfOwnerByIndex"].encodeAbi(address, index)
+  }, "latest"]
+  let response = status.callPrivateRPC("eth_call", payload)
+  let res = parseJson($response)["result"].str
+  if (res == "0x"):
+    return -1
+  result = fromHex[int](res)
+
+proc tokensOfOwnerByIndex(contract: Contract, address: EthAddress): seq[int] =
+  var index = 0
+  var token: int
+  result = @[]
+  while (true):
+    token = tokenOfOwnerByIndex(contract, address, index)
+    if (token == -1 or token == 0):
+      return result
+    result.add(token)
+    index = index + 1
+
+proc getCryptoKitties*(address: EthAddress): seq[Collectible] =
+  result = @[]
+  try:
+    # TODO handle offset (recursive method?)
+    # Crypto kitties has a limit of 20
+    let url: string = fmt"https://api.cryptokitties.co/kitties?limit=20&offset=0&owner_wallet_address={$address}&parents=false"
+    let client = newHttpClient()
+    client.headers = newHttpHeaders({ "Content-Type": "application/json" })
+
+    let response = client.request(url)
+    let kitties = parseJson(response.body)["kitties"]
```

Comment:
Would be great to use json deserialization here: https://github.com/status-im/nim-status-client/blob/365e5ab4c1871f656f9754db07ccb59c60763061/src/status/libstatus/accounts.nim#L26-L27